packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1445,10 +1445,22 @@ export class KeyringController extends BaseController<
    * @returns Promise resolving when the operation completes.
    */
   async submitPassword(password: string): Promise<void> {
-    return this.#withRollback(async () => {
+    await this.#withRollback(async () => {
       this.#keyrings = await this.#unlockKeyrings(password);
       this.#setUnlocked();
     });
+
+    try {
+      // If there are stronger encryption params available, we
+      // can attempt to upgrade the vault encryption params.
+      await this.#withRollback(async () =>
+        this.#upgradeVaultEncryptionParams(),
+      );
+    } catch (error) {
+      // We don't want to throw an error if the upgrade fails
+      // since the controller is already unlocked.
+      console.error('Failed to upgrade vault encryption params:', error);
+    }
```

Comment:
I think that consumers listening to `:changeState` and `:unlock` events would still execute normally even if this isn't wrapped in a `try catch`, but if a consumer relies on the promise returned by the function it would fail without suppressing the error

Reply:
Actually, messenger listeners will be executed before the vault rollback is even attempted